File: manuelalex/src/statements/IfElseStatement.js
/**
 * Created by Manuel on 13/02/2017.
 */

import {IfStatement}  from './IfStatement.js';

export class IfElseStatement extends IfStatement {

    constructor(condition, ifBody, elseBody, location) {
        super(condition, ifBody, elseBody, location);
        this.elseBody = elseBody;

    }

    getElseBody(){
        return this.elseBody;
    }


    accept(visitor){
        visitor.visitIfElseStatement(this);
    }

    render(visitor, view){
        visitor.renderIfElseStatement(this, view);
    }

}

File: koopmans/lib/ql/ast/statement/if_else_statement.rb
module QL
  module AST
    class IfElseStatement
      attr_reader :if_body, :else_body, :condition

      def initialize(condition, if_body, else_body)
        @condition = condition
        @if_body = if_body
        @else_body = else_body
      end

      def accept(visitor, argument = nil)
        visitor.visit_if_else_statement(self, argument)
      end
    end
  end
end
